src/muse/quantities.py

```diff
@@ -164,18 +164,25 @@ def gross_margin(
     var_exp = tech.var_exp
     fixed_outputs = tech.fixed_outputs
     fixed_inputs = tech.fixed_inputs
-
-    # Hours ratio
-    variable_costs = convert_timeslice(
-        var_par * capacity**var_exp, prices.timeslice, QuantityType.EXTENSIVE
-    )
-
-    prices = prices.sel(region=capacity.region).interp(year=capacity.year)
+    # We separate the case where we have one or more regions
+    caparegions = np.array(capacity.region.values).reshape(-1)
+    if len(caparegions) > 1:
+        prices.sel(region=capacity.region)
+    else:
+        prices = prices.where(prices.region == capacity.region, drop=True)
+    prices = prices.interp(year=capacity.year.values)
 
     # Filters
     environmentals = is_pollutant(technologies.comm_usage)
     enduses = is_enduse(technologies.comm_usage)
 
+    # Hours ratio
+    variable_costs = convert_timeslice(
```

Comment:
It migth be worth including in the docstring the equation that this represents, so the code is easier to understand. Even better if a reference to a paper or web is included giving more context.

Reply:
Thanks. I added the links and added explanations.
One thing I corrected, is that it is more common to define the gross margin as a percentage of the revenues. Therefore I added these extra steps.